gno.land/cmd/gnoland/genesis_download.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/gnolang/gno/tm2/pkg/commands"
+)
+
+var (
+	errNoChainID         = errors.New("no chain ID specified")
+	errChainNotSupported = errors.New("chain ID not supported")
+)
+
+const test4ID = "test4"
+
+const deploymentPathFormat = "https://raw.githubusercontent.com/gnolang/gno/refs/heads/master/misc/deployments/%s.gno.land/genesis.json"
+
+var genesisSHAMap = map[string]string{
+	test4ID: "beb781dffc09b96e3114fb7439fa85c4fe8ea796f64ec0cd3801a6b518ab023c",
+}
```

Comment:
This should be automated; or at least we should test the verification (ie. each source should map the sha sum, this should be an explicit test)

Also the link to genesis.json shouldn't be master, but a permalink like a commit. I think we can avoid making this dynamic with the `%s`, but instead have a struct with `Source string; Checksum string`

Reply:
I adopted the metadata structure you were suggesting:

[e3fb205](https://github.com/gnolang/gno/pull/2841/commits/e3fb2051fa2ae12ddc11a382ac7857acaf87a178)

Our unit tests already cover actually fetching the genesis from the source and verifying it